src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GenericLookupResult.cs

```diff
@@ -481,10 +487,7 @@ public override ISymbolNode GetTarget(NodeFactory factory, Instantiation typeIns
             if (factory.Target.Abi == TargetAbi.CoreRT)
             {
                 MethodDesc instantiatedMethod = _method.InstantiateSignature(typeInstantiation, methodInstantiation);
-
-                // https://github.com/dotnet/corert/issues/2342 - we put a pointer to the virtual call helper into the dictionary
-                // but this should be something that will let us compute the target of the dipatch (e.g. interface dispatch cell).
-                return factory.ReadyToRunHelper(ReadyToRunHelperId.VirtualCall, instantiatedMethod);
+                return factory.InterfaceDispatchCell(instantiatedMethod);
             }
```

Comment:
Ok, I'll buy it.

---
In reply to: [111038103](https://github.com/dotnet/corert/pull/2928#discussion_r111038103) [](ancestors = 111038103)